kernel/futex.c

```diff
@@ -0,0 +1,130 @@
+/*
```

Comment:
Would it make sense to place all file under CONFIG_USERSPACE?

Reply:
It is, look at target_sources_ifdef(CONFIG_USERSPACE, ...) in CMakeLists.txt

Reply:
@andrewboie that's actually what I had checked, so I was confused, because inside this file there are some CONFIG_USERSPACE #ifdefs.

Reply:
oh I see, yes @wentongwu please remove those